### PR TITLE
#2208: Dependency upgrades epic

### PR DESCRIPTION
**Pull request by samrichca**
# Description of changes

- update to React 17, Next 12, Apollo 3
- remove unused testing dependencies
- revise Babel plugins
- see commit list for specific issues

## Type of Change

- [ ] Bug
- [ ] Styling
- [x] New Feature

## Checklist before requesting review

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -10,8 +10,5 @@
     ],
     ["@emotion/babel-preset-css-prop", { "autoLabel": true }]
   ],
-  "plugins": [
-    "@babel/plugin-proposal-optional-chaining",
-    "@babel/plugin-proposal-object-rest-spread"
```
Comment (samrichca) [inline .babelrc:15]: rest-spread is included with Next

File: .gitignore [hunk 2]
```diff
@@ -60,6 +60,7 @@ typings/
 # next.js build output
 .next
 out/
+next-env.d.ts
```
Comment (samrichca) [inline .gitignore:63]: new version of Next is adding generated files

File: Jenkinsfile.groovy [hunk 3]
```diff
@@ -75,9 +75,6 @@ spec:
                     sh "npm ci"
                     sh "npm run test"
                 }
-                container('node') {
```
Comment (samrichca) [inline Jenkinsfile.groovy:78]: we're running gql validation during development now

File: components/ApplicationRoot.tsx [hunk 4]
```diff
@@ -17,8 +17,7 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import ApolloClient from 'apollo-client';
-import { ApolloLink } from 'apollo-link';
+import { ApolloClient, ApolloLink, ApolloProvider } from '@apollo/client';
```
Comment (samrichca) [inline components/ApplicationRoot.tsx:20]: Apollo is now an all-in-one library

**Review by joneubank — APPROVED** [review]

**Pull request merged**